pkg/controller/security/manager.go

```diff
@@ -268,7 +217,6 @@ func (s *SecretManager) deleteCert(identity string) {
 	log.Debugf("remove identity: %v refCnt : %v", identity, certificate.refCnt)
 	if certificate.refCnt == 0 {
 		delete(s.certsCache.certs, identity)
-		s.certsRotateQueue.delete(identity)
```

Comment:
If no delete operation here, does it mean that the certificate will only be automatically deleted when it expires?
So do we no longer need to consider the delayed deletion logic in the case of pod restarts?

Reply:
It is intended to let it exist, we do check whether we  should sign for a identity when we do sign.

I would double check the case at a time the identity refcnt is 0. And at a future time it is added. 

I will report back

Reply:
not a problem now